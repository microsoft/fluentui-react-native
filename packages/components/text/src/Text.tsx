import { textName, ITextType } from './Text.types';
import { compose, IComposeTypecast } from '@uifabricshared/foundation-compose';
import { Text as RNText } from 'react-native';
import { filterTextProps, ITextPropsWin32 } from '@fluentui-react-native/adapters';
import { foregroundColorTokens, textTokens } from '@fluentui-react-native/tokens';
import { settings } from './Text.settings';

export const Text = compose<ITextType>({
  displayName: textName,
  settings,
  slots: {
    root: { slotType: RNText, filter: filterTextProps }
  },
  styles: {
    root: [textTokens, foregroundColorTokens]
  }
});

export const TextWin32 = Text as IComposeTypecast<ITextType<ITextPropsWin32>>;

export default Text;
