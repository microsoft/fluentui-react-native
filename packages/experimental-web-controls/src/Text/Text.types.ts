import { IComponent, IRenderData } from '@uifabric/foundation-compose';
import { IComponentSettings, IStyleProp } from '@uifabric/theme-settings';
import { ITextTokens, IForegroundColorTokens } from '../tokens/index';
import { ICSSStyle } from '../htmlTypes';

/**
 * Properties for fabric native text field, these extend the default props for text
 */
export interface ITextProps extends ITextTokens, IForegroundColorTokens {
  disabled?: boolean;
  children?: string;
  style?: IStyleProp<ICSSStyle>;
}

export interface ITextSlotProps {
  root: ITextProps;
}

export type ITextSettings = IComponentSettings<ITextSlotProps>;
export type ITextComponent = IComponent<ITextProps, ITextSettings>;
export type ITextRenderData = IRenderData<ITextProps, ITextSettings>;
