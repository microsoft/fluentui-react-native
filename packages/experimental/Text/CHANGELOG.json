{
  "name": "@fluentui-react-native/experimental-text",
  "entries": [
    {
      "date": "Fri, 18 Jun 2021 00:38:19 GMT",
      "tag": "@fluentui-react-native/experimental-text_v0.5.1",
      "version": "0.5.1",
      "comments": {
        "patch": [
          {
            "comment": "Apply prettier to experimental, run",
            "author": "ruaraki@microsoft.com",
            "commit": "907fea51098ba59a8235128e17fb82a327b7adb2",
            "package": "@fluentui-react-native/experimental-text"
          }
        ]
      }
    },
    {
      "date": "Tue, 15 Jun 2021 00:55:36 GMT",
      "tag": "@fluentui-react-native/experimental-text_v0.5.0",
      "version": "0.5.0",
      "comments": {
        "minor": [
          {
            "comment": "Add support for ColorValue",
            "author": "ruaraki@microsoft.com",
            "commit": "90e0dcb92a4ea4ee5ea5373f2c54defa127749bf",
            "package": "@fluentui-react-native/experimental-text"
          }
        ]
      }
    },
    {
      "date": "Thu, 15 Apr 2021 20:08:17 GMT",
      "tag": "@fluentui-react-native/experimental-text_v0.4.1",
      "version": "0.4.1",
      "comments": {
        "none": [
          {
            "comment": "Merge remote-tracking branch 'upstream/master' into amgleitman/icon-macos",
            "author": "adam.gleitman@gmail.com",
            "commit": "06834001f0f0238d2a7ad99a51a430b28eaa8daa",
            "package": "@fluentui-react-native/experimental-text"
          }
        ]
      }
    },
    {
      "date": "Sat, 06 Mar 2021 00:20:05 GMT",
      "tag": "@fluentui-react-native/experimental-text_v0.4.0",
      "version": "0.4.0",
      "comments": {
        "patch": [
          {
            "comment": "Base theme on macOS changed to apple theme",
            "author": "saadnajmi2@gmail.com",
            "commit": "1b3d9f6ef10c1138b6f9701b5edf9ef52be623e6",
            "package": "@fluentui-react-native/experimental-text"
          },
          {
            "comment": "Use ColorValue from @fluentui-react-native/theme-types",
            "author": "saadnajmi2@gmail.com",
            "commit": "71b29ea2c7c28d3cdadbade15f2b2c60127e48ec",
            "package": "@fluentui-react-native/experimental-text"
          }
        ],
        "minor": [
          {
            "comment": "Update to react-native 0.63",
            "author": "30809111+acoates-ms@users.noreply.github.com",
            "commit": "d35d3b45f12a670648edaa7143b210ed43360996",
            "package": "@fluentui-react-native/experimental-text"
          }
        ]
      }
    },
    {
      "date": "Mon, 01 Mar 2021 22:53:12 GMT",
      "tag": "@fluentui-react-native/experimental-text_v0.3.2",
      "version": "0.3.2",
      "comments": {
        "patch": [
          {
            "comment": "Update deps",
            "author": "warleu@microsoft.com",
            "commit": "23cd3d437e63e105f5feee16a1b8ef8e769b61c1",
            "package": "@fluentui-react-native/experimental-text"
          }
        ]
      }
    },
    {
      "date": "Wed, 13 Jan 2021 20:04:26 GMT",
      "tag": "@fluentui-react-native/experimental-text_v0.3.0",
      "version": "0.3.0",
      "comments": {
        "minor": [
          {
            "comment": "update FUA version for button",
            "author": "67026167+chiuam@users.noreply.github.com",
            "commit": "a0b6666ced16ccb8d06e7f91dc3136294d209de7",
            "package": "@fluentui-react-native/experimental-text"
          }
        ]
      }
    },
    {
      "date": "Thu, 17 Dec 2020 00:09:27 GMT",
      "tag": "@fluentui-react-native/experimental-text_v0.2.2",
      "version": "0.2.2",
      "comments": {
        "none": [
          {
            "comment": "Add readme for svg",
            "author": "warleu@microsoft.com",
            "commit": "c2c28c356aabbd3e74a8d085015a3f4507fbdbe9",
            "package": "@fluentui-react-native/experimental-text"
          }
        ]
      }
    },
    {
      "date": "Fri, 23 Oct 2020 22:27:37 GMT",
      "tag": "@fluentui-react-native/experimental-text_v0.2.0",
      "version": "0.2.0",
      "comments": {
        "minor": [
          {
            "comment": "RNIcon feature",
            "author": "warleu@microsoft.com",
            "commit": "83a0f448bf9ee9dddece08e7c277304dcef96388",
            "package": "@fluentui-react-native/experimental-text"
          }
        ]
      }
    },
    {
      "date": "Thu, 22 Oct 2020 21:21:59 GMT",
      "tag": "@fluentui-react-native/experimental-text_v0.1.2",
      "version": "0.1.2",
      "comments": {
        "none": [
          {
            "comment": "onDismiss not called when contextual menu is dismissed with item click",
            "author": "lehon@microsoft.com",
            "commit": "4f3ed9d5555e86a6017901aa2f010732d6d6fe5a",
            "package": "@fluentui-react-native/experimental-text"
          }
        ]
      }
    },
    {
      "date": "Mon, 19 Oct 2020 22:06:02 GMT",
      "tag": "@fluentui-react-native/experimental-text_v0.1.2",
      "version": "0.1.2",
      "comments": {
        "patch": [
          {
            "comment": "rework theme usage in tester",
            "author": "jasonmo@microsoft.com",
            "commit": "5657a340608ad2e1fb6b7e61f92d706e350847b1",
            "package": "@fluentui-react-native/experimental-text"
          }
        ]
      }
    },
    {
      "date": "Mon, 12 Oct 2020 04:09:31 GMT",
      "tag": "@fluentui-react-native/experimental-text_v0.1.1",
      "version": "0.1.1",
      "comments": {
        "patch": [
          {
            "comment": "new variants",
            "author": "email not defined",
            "commit": "d1c543f21301b22a84fb6968a2c2612bb487be39",
            "package": "@fluentui-react-native/experimental-text"
          }
        ]
      }
    }
  ]
}
