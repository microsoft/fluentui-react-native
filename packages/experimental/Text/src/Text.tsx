/** @jsx withSlots */
import {
  compose,
  FontTokens,
  FontVariantTokens,
  UseSlots,
  buildProps,
  fontStyles,
  mergeProps,
  withSlots,
  ITheme
} from '@fluentui-react-native/experimental-framework';
import { Text as RNText } from 'react-native';
import { filterTextProps, ITextProps } from '@fluentui-react-native/adapters';

const textName = 'Text';

/**
 * Text tokens, these are the internally configurable values for Text elements. In particular these
 * drive decisions on how to build the styles
 */
export interface TextTokens extends FontTokens {
  /** foreground text color */
  color?: string;

  /** alternate mode for disabled look and feel */
  disabled?: TextTokens;
}

/**
 * Text props, based off of the standard react-native TextProps with some new extensions
 */
export type TextProps<TBase = ITextProps> = TBase &
  FontVariantTokens & {
    /** foreground text color */
    color?: string;

    /** whether or not this text should be presented as disabled */
    disabled?: boolean;
  };

/**
 * These are the tokens which are also present in props. If specified in props this will override the values
 * from tokens.
 */
const tokenProps: (keyof TextTokens)[] = ['variant', 'color'];

/** Type to use for compose */
interface TextType {
  props: TextProps;
  slotProps: { root: ITextProps };
  tokens: TextTokens;
}

export const Text = compose<TextType>({
  displayName: textName,
  /** Settings for the use-styling hook */
  tokens: [
    t => ({
      variant: 'secondaryStandard',
      color: t.colors.bodyText,
      disabled: {
        color: t.colors.disabledText
      }
    }),
    textName
  ],
  states: ['disabled'],
  tokenProps,
  slotProps: {
    root: buildProps<ITextProps, TextTokens>(
      (tokens: TextTokens, theme: ITheme) => ({
        style: {
          margin: 0,
          color: tokens.color,
          ...fontStyles.from(tokens, theme)
        }
      }),
      ['color', ...fontStyles.keys]
    )
  },
  /** Settings for the useSlots that will be passed on */
  slots: { root: RNText },
  filters: { root: filterTextProps },
  /** render function for the component */
  render: (props: TextProps, useSlots: UseSlots<TextType>) => {
    // stage one, execute any hooks, styling lookups to build the styled slot
    const Root = useSlots(props).root;
    // return a function used to complete the render
    return (rest: TextProps, children: React.ReactNode) => <Root {...mergeProps(props, rest)}>{children}</Root>;
  }
});

export default Text;
