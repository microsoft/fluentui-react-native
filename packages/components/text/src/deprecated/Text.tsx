import { textName, ITextType } from './Text.types';
import { compose } from '@uifabricshared/foundation-compose';
import { Text as RNText } from 'react-native';
import { filterTextProps } from '@fluentui-react-native/adapters';
import { foregroundColorTokens, textTokens } from '@fluentui-react-native/tokens';
import { settings } from './Text.settings';

export const Text = compose<ITextType>({
  displayName: textName,
  settings,
  slots: {
    root: { slotType: RNText, filter: filterTextProps },
  },
  styles: {
    root: [textTokens, foregroundColorTokens],
  },
});

export default Text;
