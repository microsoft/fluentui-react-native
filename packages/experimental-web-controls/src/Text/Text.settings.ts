import { ITextSlotProps } from './Text.types';
import { IComposeSettings } from '@uifabricshared/foundation-compose';

export const settings: IComposeSettings<ITextSlotProps> = [
  {
    root: {
      fontFamily: 'primary',
      fontSize: 'medium',
      fontWeight: 'medium',
      color: 'bodyText',
      style: {
        margin: 0
      }
    },
    _overrides: {
      disabled: {
        root: {
          style: {
            color: 'bodyTextDisabled'
          }
        }
      }
    },
    _precedence: ['disabled']
  },
  'RNFText'
];
