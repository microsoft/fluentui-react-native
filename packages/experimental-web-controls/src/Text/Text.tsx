import { ITextComponent, ITextRenderData, ITextProps } from './Text.types';
import { compose } from '@uifabric/foundation-compose';
import { textTokenKeys, foregroundColorKeys, processTextTokens, processForegroundTokens } from '../tokens/index';
import { mergeSettings } from '@uifabric/theme-settings';
import { loadTextSettings } from './Text.settings';

loadTextSettings();

export const Text = compose<ITextComponent>({
  className: 'RNFText',
  tokenProcessors: [
    {
      keyProps: (textTokenKeys as (keyof ITextProps)[]).concat(foregroundColorKeys),
      processor: (keyProps: ITextProps, data: ITextRenderData) => {
        const toMerge = { root: {} };
        processTextTokens(keyProps, toMerge.root);
        processForegroundTokens(keyProps, toMerge.root);
        return mergeSettings(data.slotProps, toMerge);
      }
    }
  ],
  slots: {
    root: 'div'
  }
});

export default Text;
