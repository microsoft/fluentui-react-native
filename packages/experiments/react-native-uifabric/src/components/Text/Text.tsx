import { textName, ITextType } from './Text.types';
import { compose } from '@uifabricshared/foundation-compose';
import { Text as RNText } from 'react-native';
import { filterTextProps } from '../../utilities/RenderHelpers';
import { foregroundColorTokens, textTokens } from '../../tokens';
import { settings } from './Text.settings';

export const Text = compose<ITextType>({
  displayName: textName,
  settings,
  slots: {
    root: { slotType: RNText, filter: filterTextProps }
  },
  styles: {
    root: [textTokens, foregroundColorTokens]
  }
});

export default Text;
