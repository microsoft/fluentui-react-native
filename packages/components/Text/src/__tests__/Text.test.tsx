import * as React from 'react';

import * as renderer from 'react-test-renderer';

import { Text } from '../Text';

describe('Text component tests', () => {
  it('Text default', () => {
    const tree = renderer.create(<Text>Text default</Text>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Text all props', () => {
    const tree = renderer.create(<Text variant="bodyStandard">All props</Text>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Text all tokens', () => {
    const BoldText = Text.customize({
      fontFamily: 'Wingdings',
      fontWeight: '900',
      fontSize: 20,
    });
    const tree = renderer.create(<BoldText>All tokens</BoldText>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Text variants render correctly with style', () => {
    const style = {
      marginBottom: 8,
      marginTop: 4,
    };
    const tree = renderer.create(
      <Text variant="heroLargeSemibold" color="blue" style={style}>
        Header Text
      </Text>,
    );
    expect(tree).toMatchSnapshot();
  });
});
