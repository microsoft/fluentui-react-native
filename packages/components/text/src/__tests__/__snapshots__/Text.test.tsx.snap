// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text component tests Text all props 1`] = `
<Text
  ellipsizeMode="tail"
  numberOfLines={0}
  style={
    {
      "color": "#323130",
      "fontFamily": "Segoe UI",
      "fontSize": 14,
      "fontStyle": undefined,
      "fontWeight": "400",
      "letterSpacing": undefined,
      "lineHeight": undefined,
      "margin": 0,
      "textAlign": undefined,
      "textDecorationLine": undefined,
    }
  }
>
  All props
</Text>
`;

exports[`Text component tests Text all tokens 1`] = `
<Text
  ellipsizeMode="tail"
  numberOfLines={0}
  style={
    {
      "color": "#323130",
      "fontFamily": "Wingdings",
      "fontSize": 20,
      "fontStyle": undefined,
      "fontWeight": "900",
      "letterSpacing": undefined,
      "lineHeight": undefined,
      "margin": 0,
      "textAlign": undefined,
      "textDecorationLine": undefined,
    }
  }
>
  All tokens
</Text>
`;

exports[`Text component tests Text default 1`] = `
<Text
  ellipsizeMode="tail"
  numberOfLines={0}
  style={
    {
      "color": "#323130",
      "fontFamily": "Segoe UI",
      "fontSize": 13,
      "fontStyle": undefined,
      "fontWeight": "400",
      "letterSpacing": undefined,
      "lineHeight": undefined,
      "margin": 0,
      "textAlign": undefined,
      "textDecorationLine": undefined,
    }
  }
>
  Text default
</Text>
`;

exports[`Text component tests Text variants render correctly with style 1`] = `
<Text
  ellipsizeMode="tail"
  numberOfLines={0}
  style={
    {
      "color": "blue",
      "fontFamily": "Segoe UI",
      "fontSize": 60,
      "fontWeight": "600",
      "margin": 0,
      "marginBottom": 8,
      "marginTop": 4,
    }
  }
>
  Header Text
</Text>
`;
